### ISSUE TITLE
lost+found folder appears post sudo reboot if ContentManager.allow_hidden == True

### ISSUE THREAD

**Issue by guomanmin** [jupyterlab/jupyterlab#16517]
Description
When c.ContentsManager.allow_hidden = True is set in .jupyter/jupyter_server_config.py, an unintended lost+found folder appears after executing sudo reboot. This issue disrupts the user experience for those relying on hidden files, as they open a Jupyter notebook and encounter the unexpected lost+found folder.

Reproduce

- Configure c.ContentsManager.allow_hidden = True in .jupyter/jupyter_server_config.py.
- Execute sudo reboot on the system.
- Open JupyterLab.
- Observe the unexpected lost+found folder appearing.

Expected behavior
The lost+found folder should not be visible in JupyterLab when hidden files are allowed. Users should only see the relevant hidden files necessary for their work.

Please advise if there's any workaround for this bug. Thanks in advance!

**Comment by guomanmin** [comment]
Would love to get an update please :) thank you!

**Comment by krassowski** [comment]
> Users should only see the relevant hidden files necessary for their work.

How can one define _relevant_ hidden files? Would someone be willing to curate a list of files hidden even if hidden files are shown (presumably different for each operating system), resolve disputes between users having different opinions on what should be shown and what should not, and and then deal with bug reports from users who expected all hidden files to be shown but did not see a file because it was on that list? Do you know of any editor or file browser that does something like this and has such a list?

Given the above questions I am not sure if a volunteer driven project could maintain this unless there is a strong demand and community contribution (happy to be proven wrong!). However, if you would only like to ask for a permanent, user-configurable filter, this would probably be much easier to accept a PR for such an enhancement.

**Comment by JasonWeill** [comment]
@guomanmin Thank you for your contribution! The `lost+found` folder is generated on UNIX-like OSes, and is outside the control of JupyterLab. JupyterLab's concept of "hidden" files doesn't discriminate; we don't block specific hidden files, although we might consider an enhancement to add an option of blocked filenames.

Another workaround is to run JupyterLab with a server root that is not the same as file system root. From my experience, `lost+found` is typically created at the root of a file system, so if you run JupyterLab from another directory, you wouldn't see the `lost+found` directory.

@krassowski Should we keep this open as an enhancement to create a settings option to block certain filenames from showing up even if hidden files shown?

**Comment by krassowski** [comment]
Yes, I think so. I renamed and relabelled this issue. For reference, VSCode has `files.exclude` setting taking regular wildcard patterns https://stackoverflow.com/questions/30140112/how-to-hide-specified-files-directories-e-g-git-in-the-sidebar-vscode

PyCharm and other JetBrains IDEs have a concept of marking directories as excluded (often used to avoid indexing data files), and these can be hidden from the file tree (https://stackoverflow.com/questions/26839898/hide-excluded-folders-from-tree-in-intellij-idea)

**Comment by guomanmin** [comment]
Thanks for the response. I believe there is a misunderstanding here - this is not a request to block specific hidden files. I'd like to understand why lost+found folder was generated from allow_hidden being set to true and if there is a workaround, based on the assumption that lost+found folder, generated by UNIX-like OSes, is not a hidden folder, since the folder is visible via `ls` without having to `ls -a`.

**Comment by guomanmin** [comment]
Thanks for the response. I believe there is a misunderstanding here - this is not a request to block specific hidden files. I'd like to understand why lost+found folder was generated from allow_hidden being set to true and if there is a workaround, based on the assumption that lost+found folder, generated by UNIX-like OSes, is not a hidden folder, since the folder is visible via `ls` without having to `ls -a`.

**Comment by JasonWeill** [comment]
The `lost+found` folder is created by the file system check (`fsck`) utility when it detects corrupted files. I don't see any special treatment of this folder in Jupyter or JupyterLab code in any repository. I created this folder under my JupyterLab root and, with allow_hidden set to false (the default) I see it in the file browser, treated like any other non-hidden folder.

**Comment by guomanmin** [comment]
I found that the lost+found folder become visible/hidden as I set c.ContentsManager.allow_hidden to true/false, but its creation timestamp remains the same, which is indicating that it is not generated by the config change...

**Comment by krassowski** [comment]
Again, Jupyter that switch simply toggles whether to show files reported as hidden by your operating system. You can see the code which asks the operating system which files are hidden here: https://github.com/jupyter/jupyter_core/blob/fa513c1550bbd1ebcc14a4a79eb8c5d95e3e23c9/jupyter_core/paths.py#L368-L510

**Comment by krassowski** [comment]
jupyter-server's contents manager has [`ContentsManager.hide_globs`](https://jupyter-server.readthedocs.io/en/latest/api/jupyter_server.services.contents.html#jupyter_server.services.contents.manager.ContentsManager.hide_globs) attribute and [configuration option](https://jupyter-server.readthedocs.io/en/latest/other/full-config.html).

> Default: ['__pycache__', '*.pyc', '*.pyo', '.DS_Store', '*.so', '*.dyl...
> Glob patterns to hide in file and directory listings.

Since this is already supported by jupyter-server I think we can either close it or keep in case if anyone would be willing to open a PR adding a very brief documentation to the user-facing documentation (basically with a link back to jupyter-server config).
